File: 2196-create-binary-tree-from-descriptions/2196-create-binary-tree-from-descriptions.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def createBinaryTree(self, descriptions):
        children = set()
        m = {}
        for p,c,l in descriptions:
            np = m.setdefault(p, TreeNode(p))
            nc = m.setdefault(c, TreeNode(c))
            if l:
                np.left = nc
            else:
                np.right = nc
            children.add(c)
        root = (set(m) - set(children)).pop()
        return m[root]

File: 2196-create-binary-tree-from-descriptions/2196-create-binary-tree-from-descriptions.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* createBinaryTree(vector<vector<int>>& descriptions) {
        map<int, TreeNode*> mp;
    for (auto it : descriptions) {
      TreeNode* parentNode, * childNode;

      // Finding the parent Node
      if (mp.find(it[0]) != mp.end()) {
        parentNode = mp[it[0]];
      }
      else {
        parentNode = new TreeNode(it[0]);
        mp[it[0]] = parentNode;
      }

      // Finding the child Node
      if (mp.find(it[1]) != mp.end()) {
        childNode = mp[it[1]];
      }
      else {
        childNode = new TreeNode(it[1]);
        mp[it[1]] = childNode;
      }

      // Making the Edge Between parent and child Node
      if (it[2] == 1) {
        parentNode->left = childNode;
      }
      else {
        parentNode->right = childNode;
      }
    }

    // Store the childNode 
    map<int, int> storeChild;
    for (auto it : descriptions) {
      storeChild[it[1]] = 1;
    }
    // Find the root of the Tree
    TreeNode* root;
    for (auto it : descriptions) {
      if (storeChild.find(it[0]) == storeChild.end()) {
        root = mp[it[0]];
      }
    }
    return root;

        
    }
};